### PR TITLE
New old grab system and warrior lunge.

### PR DESCRIPTION
**Pull request by Binbers**
## About The Pull Request
Hello. Please refer to these two links for full details on the changes.
https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2849 and https://gitlab.com/cmdevs/colonial-warfare/-/merge_requests/2561

## Why It's Good For The Game

Returning and un-simplifying the variable grab levels and resist chances will provide plenty of RP potential as it had been as well as highlighting the teamwork aspect requiring humans to rely on each other furthermore OR understand the factors of aggressive grab in escalated RP situations.

As for the warrior change, this would distinguish the warrior caste as an individual caste it once was as opposed to being the average "click or macro combo until marine horizontal" clone of other castes that often times can't seem to decide how it wants to be balanced. It would slightly emphasize interesting and specialized gameplay over speed & competition and provide a separate medium for balancing the warrior instead of changing its health/armor/damage values upside-down every 3 months.

## Changelog

:cl:
balance: Different levels of grab offer different chances for successful resisting and different CQC skillsets offer different speeds for attempting resists

balance: Choke grab now deals less oxygen damage per ticks

balance: Choke grab will not pull the puller and pullee onto the same time anymore (because you don't fireman carry someone you're choking on your shoulders). Only carry grab will now overlap the puller and pullee onto the same tile

fix: Warrior lunge grab level finally fixed from unintentionally being passive grab. GRAB_NECK level is re-added and explicit to warriors

fix: Warrior lunge can now move pullees again

fix: Choke grab can now move mobs over to other tiles
/:cl:

### CODE REVIEW

**Comment by usnpeepoo** [comment]
The last time this was merged (before the tile click fix) marines who were lunged by warriors were stunned for an abnormal amount of time if they couldn’t resist out of it, predators sometimes getting one lunge devoured into a capture. Is this MR really just bringing back old warriors stun? The current lunge is pretty fine and balanced with what you get out of it but making it RNG will just be a massive buff to warrior.

**Comment by stanalbatross** [comment]
I'll look at this tomorrow

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -190,7 +190,7 @@
 
 		var/pull_dir = get_dir(src, pulling)
 
-		if(grab_level >= GRAB_CARRY)
+		if(grab_level == GRAB_CARRY)
 			switch(grab_level)
 				if(GRAB_CARRY)
```
Comment (stanalbatross) [inline code/modules/mob/living/living.dm:195]:
```suggestion
```
already a check for that above as you added

File: code/modules/mob/living/living_verbs.dm [hunk 2]
```diff
@@ -13,7 +13,21 @@
...
+		if(SKILL_CQC_TRAINED)
+			next_move = world.time + 18
+		if(SKILL_CQC_SKILLED)
+			next_move = world.time + 10
+		if(SKILL_CQC_EXPERT)
+			next_move = world.time + 5
+		if(SKILL_CQC_MASTER)
+			next_move = world.time
+		if(SKILL_CQC_MAX)
+			next_move = world.time
```
Comment (stanalbatross) [inline code/modules/mob/living/living_verbs.dm:30]: can be condensed into one with a comma like 
```suggestion
		if(SKILL_CQC_MASTER, SKILL_CQC_MAX)
			next_move = world.time
```
also MASTER and MAX go to the same number so it's not even needed

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Review by stanalbatross — CHANGES_REQUESTED** [review]
also you should not be trying to merge your master into dev, make a separate branch for each PR

**Comment by Binbers** [comment]
> also you should not be trying to merge your master into dev, make a separate branch for each PR

I stuck w/ master cause I chose to. If I know I'll be frequenting the code often enough to make more than one PRs at a time, sure thing I would've done so

**Comment by Binbers** [comment]
Ridding merely "if(GRAB_CARRY)" nuked the entire if else chain, fixed it and it's more cleaned up in addition to working now

**Comment by stanalbatross** [comment]
resolve conflicts and I'll review again

**Pull request closed without merging**